.github/workflows/checks.yaml

```diff
@@ -182,3 +183,8 @@ jobs:
         run: |
           . venv/bin/activate
           pytest doc/test_messages_documentation.py
+      - name: Check documentation build and links
+        run: |
+          . venv/bin/activate
+          cd doc
+          make html
```

Comment:
We can't add `linkcheck` to readthedocs as the only allow one build target (and they don't use `make html` but `sphinx-build -b html` directly).

Thus, we need our own CI to run `linkcheck`. Takes 2 additional minutes but I think that's okay considering the time most other jobs take.